### PR TITLE
Add gh_tool CLI for GitHub repository automation

### PR DESCRIPTION
**Pull request by mattgodbolt**
This PR adds a new Python CLI tool for automating GitHub repository management tasks.

## Overview

The initial implementation provides duplicate issue detection using text similarity analysis. This is the first step toward automating repository triage tasks.

## Features

- **Click-based CLI** with subcommands for future extensibility
- **find-duplicates command** for detecting duplicate issues using text similarity
- Uses **gh CLI** for GitHub API access (no token management needed)
- Text similarity using `difflib.SequenceMatcher` (ratio-based algorithm)
- Configurable similarity threshold (default: 0.6)
- Progress bar for long-running comparisons
- Age filtering support (`--min-age` parameter)
- Standard Python src-layout with **uv** for dependency management
- **Comprehensive test suite** with pytest (integrated into CI)

## Project Structure

```
etc/scripts/gh_tool/
├── src/gh_tool/          # Main package
│   ├── cli.py            # Click-based CLI interface
│   └── duplicate_finder.py  # Core duplicate detection logic
├── tests/                # Test suite
│   └── test_duplicate_finder.py
├── docs/                 # Documentation
│   ├── TRIAGE-CRITERIA.md    # Triage guidelines from manual review
│   └── PHASE1-FINDINGS.md    # Historical analysis of 855 issues
├── pyproject.toml        # Package configuration
└── README.md             # Usage documentation
```

## Usage

```bash
cd etc/scripts/gh_tool
uv sync
uv run gh_tool find-duplicates /tmp/report.md
```

**Options:**
- `--threshold FLOAT` - Similarity threshold 0-1 (default: 0.6)
- `--state {all,open,closed}` - Issue state to check (default: open)
- `--min-age DAYS` - Only check issues older than N days (default: 0)
- `--limit INTEGER` - Maximum number of issues to fetch (default: 1000)
- `--repo TEXT` - GitHub repository in owner/repo format (default: compiler-explorer/compiler-explorer)

**Example:**
```bash
# Find high-confidence duplicates in open issues
uv run gh_tool find-duplicates /tmp/report.md --threshold 0.85

# Check all issues older than 30 days
uv run gh_tool find-duplicates /tmp/report.md --state all --min-age 30
```

## Testing

The tool includes comprehensive test coverage:
- Unit tests for similarity calculation
- Integration tests for duplicate detection
- Edge case handling (transitive grouping, age filtering, threshold sensitivity)
- Report generation validation

**Run tests:**
```bash
cd etc/scripts/gh_tool
uv run pytest -v
```

Tests are integrated into CI and run on every push.

## Documentation

- **`README.md`**: Complete usage guide with examples
- **`docs/TRIAGE-CRITERIA.md`**: Comprehensive triage guidelines developed during manual review of 22+ issues
- **`docs/PHASE1-FINDINGS.md`**: Historical analysis context from initial 855 issue review

## CI Integration

The tool is integrated into the GitHub Actions workflow:
- `uv` is installed via `astral-sh/setup-uv@v6`
- Tests run automatically on every push
- Ensures tool remains functional as codebase evolves

## Next Steps

Future enhancements planned for follow-up PRs:
- GitHub Action for automatic duplicate detection on new issues
- Additional automation tools (upstream health checker, label validator, etc.)
- Automated triage reports

## Changes in this PR

- ✅ Core duplicate detection implementation
- ✅ Comprehensive test suite (192 lines)
- ✅ CI integration
- ✅ Complete documentation
- ✅ Example triage criteria and findings

### CODE REVIEW

**Comment by mattgodbolt** [comment]
## Sample Run: Duplicate Detection Results

Ran the tool with a high confidence threshold (85%) on all 844 open issues:

```bash
uv run gh_tool find-duplicates /tmp/duplicates-sample.md --threshold 0.85
```

**Results**: Found **13 potential duplicate groups** from 355,746 comparisons

### Notable Findings

**Definite Duplicates** (need action):
- **Group 1**: #5658 and #6977 - Both "[LANGUAGE REQUEST]: Forth" (98% similar)
- **Group 3**: #5764 "Flang 17.0.X" and #5925 "Flang 17" (93% similar)
- **Group 12**: #4478 and #4479 - Both TI ARM compiler requests (85% similar)

**Similar Patterns** (less certain):
- **Group 1**: 10 different language requests all match at 98% due to identical template format
- **Group 2**: 18 compiler requests match at 95% (template similarity, not actual duplicates)

### Observations

The tool successfully identifies:
1. **True duplicates**: Forth (#5658, #6977) and Flang variations (#5764, #5925)
2. **Template-based clustering**: Issues using the same template cluster together
3. **Similar compiler/library families**: clang variants, TI compilers, etc.

### Limitations

- High similarity from template usage creates noise in lower-confidence matches
- Groups 1 and 2 show this clearly - mostly template matches, not content duplicates
- Suggests we might want to normalize/strip template boilerplate before comparison

### True Positive Rate

At 85% threshold from this run:
- **3 clear duplicate groups** requiring action (Groups 3, 11, 12 have obvious duplicates)
- **10 groups** are template artifacts or legitimately different requests

For automated detection on new issues, a **higher threshold (90-95%)** or **template normalization** would reduce false positives.

File: etc/scripts/gh_tool/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "compiler-explorer-gh-tool"
+version = "0.0.1"
+description = "GitHub automation tool for Compiler Explorer"
+requires-python = ">=3.12"
+dependencies = [
+    "click>=8.1.0",
+    "pytest>=8.0.0",
```
[the remark this answers is not in the document]
Comment (mattgodbolt) [reply etc/scripts/gh_tool/pyproject.toml:12]: nah dawg

**Comment by partouf** [comment]
This is now live
